view-generator-framework/src/main/java/org/hypertrace/core/viewgenerator/service/ViewGeneratorLauncher.java

```diff
@@ -1,58 +1,121 @@
 package org.hypertrace.core.viewgenerator.service;
+import static org.hypertrace.core.viewgenerator.service.ViewGeneratorConstants.INPUT_CLASS_CONFIG_KEY;
+import static org.hypertrace.core.viewgenerator.service.ViewGeneratorConstants.INPUT_TOPIC_CONFIG_KEY;
+import static org.hypertrace.core.viewgenerator.service.ViewGeneratorConstants.JOB_CONFIG;
+import static org.hypertrace.core.viewgenerator.service.ViewGeneratorConstants.KAFKA_STREAMS_CONFIG_KEY;
+import static org.hypertrace.core.viewgenerator.service.ViewGeneratorConstants.OUTPUT_TOPIC_CONFIG_KEY;
+import static org.hypertrace.core.viewgenerator.service.ViewGeneratorConstants.SCHEMA_REGISTRY_CONFIG_KEY;
+import static org.hypertrace.core.viewgenerator.service.ViewGeneratorConstants.VIEW_GENERATOR_CLASS_CONFIG_KEY;
 
-import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
-import org.hypertrace.core.serviceframework.PlatformService;
+import com.typesafe.config.Config;
+import java.util.Map;
+import java.util.Properties;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.errors.LogAndContinueExceptionHandler;
+import org.apache.kafka.streams.kstream.Consumed;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.Produced;
+import org.hypertrace.core.kafkastreams.framework.KafkaStreamsApp;
+import org.hypertrace.core.kafkastreams.framework.serdes.SchemaRegistryBasedAvroSerde;
+import org.hypertrace.core.kafkastreams.framework.timestampextractors.UseWallclockTimeOnInvalidTimestamp;
 import org.hypertrace.core.serviceframework.config.ConfigClient;
+import org.hypertrace.core.serviceframework.config.ConfigUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/**
- * ViewGeneratorLauncher reads enriched structured traces and converts them to Pinot records using specified Avro
- * schema specified in the target view generator class, which is specified in the config.
- */
-public class ViewGeneratorLauncher extends PlatformService {
+public class ViewGeneratorLauncher extends KafkaStreamsApp {
```

Comment:
Can kafka streams framework base class extend PlatformService or is it already doing that?

Reply:
yes the KafkaStreamsApp extends PlatformService